src/2bit.js

```diff
@@ -0,0 +1,204 @@
+  var bytes = new ReadableView(dataView);
+  var dnaSize = bytes.readUint32(),
+      nBlockCount = bytes.readUint32(),
+      nBlockStarts = bytes.readUint32Array(nBlockCount),
+      nBlockSizes = bytes.readUint32Array(nBlockCount),
+      // The masks can be quite large (~2MB for chr1) and we mostly don't care
+      // about them.  So we ignore them, but we do need to know their length.
+      maskBlockCount = bytes.readUint32();
+      // maskBlockCount maskBlockStarts = bytes.readUint32Array(maskBlockCount),
+      // maskBlockSizes = bytes.readUint32Array(maskBlockCount),
+      // reserved = bytes.readUint32();
+
+  var offset = bytes.tell() + 8 * maskBlockCount + 4;
+
+  // DNA information comes after this.
+  return {
+    numBases: dnaSize,
+    unknownBlockStarts: nBlockStarts,
+    unknownBlockLengths: nBlockSizes,
+    numMaskBlocks: maskBlockCount,
+    maskBlockStarts: [],
+    maskBlockLengths: [],
+    dnaOffsetFromHeader: offset,
+    offset: fileOffset
+  };
+}
+
+
+/**
+ * Parses the 2bit file header.
+ */
+function parseHeader(dataView: DataView): TwoBitHeader {
+  var bytes = new ReadableView(dataView);
+  var magic = bytes.readUint32();
+  if (magic != TWO_BIT_MAGIC) {
+    throw 'Invalid magic';
+  }
+  var version = bytes.readUint32();
+  if (version != 0) {
+    throw 'Unknown version of 2bit';
+  }
+  var sequenceCount = bytes.readUint32(),
+      reserved = bytes.readUint32();
+
+  var sequences: Array<FileIndexEntry> = [];
+  for (var i = 0; i < sequenceCount; i++) {
+    var nameSize = bytes.readUint8();
+    var name = bytes.readAscii(nameSize);
+    var offset = bytes.readUint32();
+    sequences.push({name, offset});
+  }
+  // hg19 header is 1671 bytes to this point
+
+  return {
+    sequenceCount,
+    reserved,
+    sequences
+  };
+}
+
+
+/**
+ * Read 2-bit encoded base pairs from a DataView into an array of 'A', 'T',
+ * 'C', 'G' strings.
+ * These are returned as an array (rather than a string) to facilitate further
+ * modification.
+ */
+function unpackDNA(dataView: DataView, startBasePair: number, numBasePairs: number): Array<string> {
+  var basePairs: string[] = [];
+  basePairs.length = dataView.byteLength * 4;  // pre-allocate
+  var basePairIdx = -startBasePair;
+  for (var i = 0; i < dataView.byteLength; i++) {
+    var packed = dataView.getUint8(i);
+    for (var shift = 6; shift >= 0; shift -= 2) {
+      var bp = BASE_PAIRS[(packed >> shift) & 3];
+      if (startBasePair >= 0) {
+        basePairs[basePairIdx] = bp;
+      }
+      basePairIdx++;
+    }
+  }
+  // Remove base pairs from the end if the sequence terminated mid-byte.
+  basePairs.length = numBasePairs;
+  return basePairs;
+}
+
+/**
+ * Change base pairs to 'N' where the SequenceRecord dictates.
+ * This modifies the basePairs array in-place.
+ */
+function markUnknownDNA(basePairs: string[], dnaStartIndex: number, sequence: SequenceRecord): Array<string> {
```

Comment:
Correct on both counts.

(I'm surprised `string[]` compiled -- I'm unclear if that's a valid syntax in Flow.)